### PR TITLE
Display validations at the top level when validate is pressed.

### PR DESCRIPTION
**Pull request by jcoyne**
Ref #311 #303

### CODE REVIEW

File: __tests__/Validator.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            },
+          ],
+        },
+      },
+    },
+  }
+})
+
+describe('validate()', () => {
+  it('when the nothing is required', () => {
```
Comment (ndushay) [inline __tests__/Validator.test.js]: typo?  "the nothing" ?  and below "the something" ?

File: src/Validator.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    // If it has items > 0 or nested objects then it's valid
+    const errors = this.getItems(property).length > 0 ? [] : [{ message: 'Required', path: [resourceTemplateId, predicate], label: propertyTemplate.propertyLabel }]
+
+    property.errors = errors
+    return errors
+
+    /*
+     * TODO Check to see if property has any keys besides 'errors' and recurse on them
+     * key.rtId.predicate (maybe property[key] is equivalent to resource)
```
Comment (ndushay) [inline src/Validator.js]: make a ticket for this and reference it here?

File: src/Validator.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    return this.newState
+  }
+
+  predicatesForResourceTemplate(resourceTemplateId) {
+    return Object.keys(this.resource[resourceTemplateId]).filter(val => val !== 'resourceURI')
+  }
+
+  /**
+   * Check and see if this property is required and if so, if it's filled in.
+   * @return {Array} a list of errors for the property
```
Comment (ndushay) [inline src/Validator.js]: or empty array if no errors?

Comment (jcoyne) [reply src/Validator.js]: Yes, that's a list of 0 errors.

**Comment by jcoyne** [comment]
@ndushay changes made.

**Pull request merged**